magma/ref.py

```diff
@@ -201,6 +237,12 @@ def anon(self):
     def parent(self):
         return self.tuple.name
 
+    def named(self) -> bool:
+        return self.parent().named()
+
+    def bound(self) -> bool:
+        return self.parent().bound()
+
```

Comment:
Could add an intermediate class for Array/Tuple ref named "RecursiveRef" that defines these methods